### ISSUE TITLE
Entire tab freezes when <choice> element added

### ISSUE THREAD

**Issue by maxrothman** [mei-friend/mei-friend#109]
Given the following minimal MEI document:
```xml
<?xml version="1.0" encoding="UTF-8"?><?xml-model href="https://music-encoding.org/schema/5.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?><?xml-model href="https://music-encoding.org/schema/5.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?><mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="5.0">
   <meiHead xml:id="m1k3svg6">
   	<fileDesc>
         <titleStmt><title>foo</title></titleStmt>
         <pubStmt></pubStmt>
      </fileDesc>
   </meiHead>
   <music xml:id="mujr4uc1">
      <body xml:id="bakr4ha">
         <mdiv xml:id="m4kn5qi">
            <score xml:id="s3nb0j8">
               <scoreDef xml:id="s1u049a">
                  <staffGrp xml:id="sj31095">
                     <staffDef xml:id="P1" lines="5" n="1" ppq="4">
                        <clef xml:id="cusgk3x" line="2" shape="G"/>
                        <keySig xml:id="k1nemqaw">
                           <choice
                           <orig>
                              <keyAccid xml:id="k10uzq7v" accid="f" pname="b"/>
                              <keyAccid xml:id="k10gm87c" accid="f" pname="e"/>
                              <add xml:id="a19bmee" hand="#wud92h4">
                                 <keyAccid xml:id="kp9nu6k" accid="s" pname="c"/>
                              </add>
                           </orig>
                           <reg>
                              <keyAccid accid="f" pname="b"/>
                              <keyAccid accid="f" pname="e"/>
                              <keyAccid accid="s" pname="c"/>
                           </reg>
                        </keySig>
                        <meterSig xml:id="m1mgc3gh" count="2" unit="4"/>
                     </staffDef>
                  </staffGrp>
               </scoreDef>
            </score>
         </mdiv>
      </body>
   </music>
</mei>
```

**To Reproduce**
1. Paste the document above into mei-friend
2. Put your cursor on line 17 before `<orig>`
3. Type `<choice`

**Expected behavior**
mei-friend shows what has been typed. In this particular case a `<choice>` is invalid in that location, so an error should be shown

**Actual behavior**
mei-friend's entire browser tab freezes and stops responding to input

**Desktop (please complete the following information):**
 - OS: MacOS 12.7.5
 - Browser: Chrome 127.0.6533.120 and .100 (tested on both)
 - mei-friend version: Version: mei-friend 1.0.14 (23 May 2024)

**Additional context**
It seems like it might be a infinite-loop-type issue, it takes a moment for the tab to freeze (after refreshing I can quickly remove the offending code before it locks up), and while frozen, my computer fans run loudly.

**Comment by maxrothman** [comment]
Update: I've found a few other cases where adding a `<choice>` causes mei-friend to freeze, for example when wrapping a note elements, so it might be that **any** `<choice>` causes the freeze, independent of location in the document.

**Comment by wergo** [comment]
Thanks for spotting this. We will look into this!

**Comment by wergo** [comment]
Fixed in 0cb210ecc. Already available online at https://mei-friend.mdw.ac.at/ and https://staging.mei-friend.mdw.ac.at/. 

Thanks again @maxrothman for spotting this bug.
